zipkin-storage/influxdb/src/main/java/zipkin2/storage/influxdb/InfluxDBSpanConsumer.java

```diff
@@ -0,0 +1,124 @@
+ * the License.
+ */
+package zipkin2.storage.influxdb;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.influxdb.InfluxDBException;
+import org.influxdb.dto.Point;
+import zipkin2.Annotation;
+import zipkin2.Call;
+import zipkin2.Endpoint;
+import zipkin2.Span;
+import zipkin2.storage.SpanConsumer;
+import org.influxdb.dto.BatchPoints;
+
+final class InfluxDBSpanConsumer implements SpanConsumer {
+  final InfluxDBStorage storage;
+
+  InfluxDBSpanConsumer(InfluxDBStorage influxDB) {
+    this.storage = influxDB;
+    try {
+      storage.get().createDatabase(storage.database());
+    } catch(InfluxDBException e){}
+  }
+
+  @Override
+  public Call<Void> accept(List<Span> spans) {
+    if (spans.isEmpty()) return Call.create(null);
+
+    BatchPoints batch = BatchPoints
+      .database(storage.database())
+      .retentionPolicy(storage.retentionPolicy())
+      .build();
+    for (Span span : spans) {
+      Point point = Point
+        .measurement(storage.measurement())
+        .tag("trace_id", span.traceId())
+        .tag("id", span.id())
+        .tag("parent_id", span.parentId() == null ? span.id() : span.parentId())
+        .tag("name", span.name())
+        .tag("service_name", serviceName(span))
+        .addField("duration_ns", span.duration() * 1000)
+        .time(span.timestamp(), TimeUnit.MICROSECONDS)
+        .build();
+      batch.point(point);
+
+      for (Annotation anno : span.annotations()) {
+        Point annoPoint = Point
+          .measurement(storage.measurement())
+          .tag("trace_id", span.traceId())
+          .tag("id", span.id())
+          .tag("parent_id", span.parentId() == null ? span.id() : span.parentId())
+          .tag("name", span.name())
+          .tag("service_name", serviceName(span))
+          .tag("endpoint_host", host(span))
+          .tag("annotation", anno.value())
+          .addField("duration_ns", span.duration() * 1000)
+          .time(span.timestamp(), TimeUnit.MICROSECONDS)
+          .build();
+        batch.point(annoPoint);
+      }
+
+      if (span.tags() != null) {
+        for (Map.Entry<String, String> tag : span.tags().entrySet()) {
+          Point taggedPoint = Point
+            .measurement(storage.measurement())
+            .tag("trace_id", span.traceId())
+            .tag("id", span.id())
+            .tag("parent_id", span.parentId() == null ? span.id() : span.parentId())
+            .tag("name", span.name())
+            .tag("service_name", serviceName(span))
+            .tag("annotation_key", tag.getKey())
+            .tag("annotation", tag.getValue())
+            .tag("endpoint_host", host(span))
+            .addField("duration_ns", span.duration() * 1000)
+            .time(span.timestamp(), TimeUnit.MICROSECONDS)
+            .build();
+          batch.point(taggedPoint);
+        }
+      }
+    }
+    storage.get().write(batch);
```

Comment:
This will synchronously write to InfluxDB (https://github.com/influxdata/influxdb-java/issues/386).
If you want to do it async, enable the batch mode when you create the client:
```Java
// Flush every 2000 Points, at least every 1000ms
influxDBClient.enableBatch(2000, 1000, TimeUnit.MILLISECONDS);
```
and use the `write` method for `Point` and not `BatchPoint`.

Reply:
We discussed a long time ago with @adriancole about this and I think it's good as it is atm. Thanks @fmachado